### PR TITLE
Tutorial: Port code for RemarkCommand

### PR DESCRIPTION
**Pull request by j-lum**
Now that the code base for AB3 has stabilized, let's port the removed tutorial from AB4 to AB3.
This should belong on its own branch. 

This commit does not contain the tutorial adoc. I'll be opening a PR to merge that into master.

### CODE REVIEW

**Comment by damithc** [comment]
I wonder if the initial steps can be even smaller. e.g., start with a simpler form of the command `remark` (not `remark 2 r/ good`). This will result in a smaller first commit.

On a related note, the current breakdown still feels like written by someone who knows the model solution, whereas the reader doesn't have that kind of expertise. Hence, the steps given may sound 'magical' to the reader (i.e., there is an empathy gap between the writer and the reader, which makes it harder for the reader to connect with the content). 
Keep in mind that we are not only training students to work with AB-3 code base, but also training them _how to use a strategy of 'tiny increments' when working in an unfamiliar code base_. The latter has more value but requires more effort in writing. For example, we should also explain the thought process in a way that can be generalized (`trace through a part that is similar to our target feature` instead of `trace through the delete command`)

**Comment by j-lum** [comment]
From the code, we start from 
1. recognizing `remark` and ignoring everything else
1. recognizing the whole argument string
1. parsing the string to obtain more useful data

So I don't think the code needs to change, maybe we can emphasize the build up a bit more in the tutorial?

**Comment by damithc** [comment]
> So I don't think the code needs to change, maybe we can emphasize the build up a bit more in the tutorial?

The first commit has more code than necessary. e.g.,
* any mention of prefix
* an elaborate help message
* test case

These are things an experienced AB programmer would include in the first step (because she knows these will be needed eventually) but not a newcomer.

**Comment by j-lum** [comment]
I'm writing a new bridging section between the intro and the first bits to code to help the students case out the problem and break it down into logical components. From there we can re-evaluate how we want to structure the commits since the old tutorial (and its code) was meant to be the model answer.

**Comment by j-lum** [comment]
Added a new intermediate step before `Logic: Teach the app to accept 'remark' but do nothing `.

The rationale was that students with little exposure to the code base isn't going to write standard-compliant code right from the code and we should ease them into the process.

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package seedu.address.logic.commands;
 
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REMARK;
```
Comment (damithc) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:4]: Not needed for this step

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 2]
```diff
@@ -9,8 +12,18 @@
 
     public static final String COMMAND_WORD = "remark";
 
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the remark of the person identified "
+            + "by the index number used in the last person listing. "
+            + "Existing remark will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_REMARK + "[REMARK]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_REMARK + "Likes to swim.";
```
Comment (damithc) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:29]: not needed for this step

**Review by damithc — CHANGES_REQUESTED** [review]
2nd step needs to be simplified further as it contains things the reader will not see the need for. Plus, it needs to match the first step: currently, the first step creates a Result object and the second step switches to the Exception route.

**Comment by damithc** [comment]
Is there a 1-1 mapping between the commits here and the sections in the tutorial? It's ideal if there is. In that case, section names and commit subjects should match closely.

**Comment by damithc** [comment]
Let's move to the next iteration

**Comment by j-lum** [comment]
Not all commit titles are strict 1-1 matches but they are close enough with some minor differences in grammar. 

Going to shove this through for tutorial on week 6.

**Pull request closed without merging**